From 4dc25f07adec55ec00a5d5a2896c6ab6427c7784 Mon Sep 17 00:00:00 2001
From: Jon Maloy <jon.maloy@ericsson.com>
Date: Mon, 16 Oct 2017 16:04:51 +0200
Subject: [PATCH 041/176] tipc: fix rebasing error

commit 36c0a9dfc6613242ba1de012e2d15145cdaae805 upstream

In commit 2f487712b893 ("tipc: guarantee that group broadcast doesn't
bypass group unicast") there was introduced a last-minute rebasing
error that broke non-group communication.

We fix this here.

Signed-off-by: Jon Maloy <jon.maloy@ericsson.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Zhenbo Gao <zhenbo.gao@windriver.com>
---
 net/tipc/link.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/net/tipc/link.c b/net/tipc/link.c
index 723dd69..870b9b8 100644
--- a/net/tipc/link.c
+++ b/net/tipc/link.c
@@ -1052,6 +1052,7 @@ static bool tipc_data_input(struct tipc_link *l, struct sk_buff *skb,
 			return true;
 		}
 	case CONN_MANAGER:
+		skb_queue_tail(inputq, skb);
 		return true;
 	case GROUP_PROTOCOL:
 		skb_queue_tail(mc_inputq, skb);
-- 
1.7.9.5

